File: src/IdentityServer4.Admin.Application/ViewModels/Client/CreateClientViewModel.cs
using IdentityServer4.Admin.Domain.Models;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace IdentityServer4.Admin.Application.ViewModels.Client
{
    public class CreateClientViewModel
    {
        [Required]
        public string ClientId { get; set; }
        [Required]
        public string ClientName { get; set; }
        public string ClientUri { get; set; }
        public string LogoUri { get; set; }
        public string Description { get; set; }
        [JsonIgnore]
        public ClientType ClientType => ClientType.From(ClientTypeId);
        public int ClientTypeId { get; set; }
    }
}

File: src/IDS4.AdminUI/src/@types/CreateClientViewModel.ts
export default class CreateClientViewModel {
    clientId?: string;
    clientName?: string;
    clientUri?: string;
    logoUri?: string;
    description?: string;
    clientTypeId?: number;

    /**
     *
     */
    constructor(clientId: string, clientName: string, clientType: number, clientUri: string, logoUri: string, description: string) {
        this.clientId = clientId;
        this.clientName = clientName;
        this.clientUri = clientUri;
        this.logoUri = logoUri;
        this.clientTypeId = clientType;
        this.description = description;
    }
}